staging/src/k8s.io/code-generator/cmd/validation-gen/validators/each.go

```diff
@@ -168,6 +168,8 @@ func (lmktv listMapKeyTagValidator) GetValidations(context Context, _ []string,
 		return Validations{}, fmt.Errorf("no field for JSON name %q", payload)
 	} else if k := realType(memb.Type).Kind; k != types.Builtin {
 		return Validations{}, fmt.Errorf("only primitive types can be list-map keys, not %s", k)
```

Comment:
Not for this PR, but-

I never checked if we allow for types that marshal to scalars are allowed as keys.  That is, could something like Quantity be a key?

Reply:
This is a great question.  What's the right way to detect that?